x-pack/test/functional_enterprise_search/services/app_search_client.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import http from 'http';
+
+/**
+ * A simple request client for making API calls to the App Search API
+ */
+const makeRequest = <T>(method: string, path: string, body?: object): Promise<T> => {
+  return new Promise(function(resolve, reject) {
+    // Throw error if no key is configured
+    // Handle API errors
```

Comment:
Is `// Handle API errors` a stray comment that got accidentally moved? I'm a little confused as to how it applies directly to the line directly below it 🤔

Reply:
I think that was supposed to say "// TODO Handle API errors"

Reply:
I'm going to address this.

Reply:
Added error handling so that when this thing bombs out on an App Search API call, it gives you enough context to fix it.

```
     Error: the string "Error calling App Search API: {\"error\":\"Invalid authentication token.\"}" was thrown, throw an Error :)


       └- ✖ fail: "Enterprise Search Engines Overview "before all" hook in "Engines Overview""
```

Reply:
Awesome, looks terrific!